third_party/airflow/armada/operators/armada.py

```diff
@@ -16,151 +16,328 @@
-
-    Airflow operators inherit from BaseOperator.
-
-    :param name: The name of the airflow task
-    :param armada_channel_args: GRPC channel arguments to be used when creating
-      a grpc channel to connect to the armada server instance.
-    :param job_service_channel_args: GRPC channel arguments to be used when creating
-      a grpc channel to connect to the job service instance.
-    :param armada_queue: The queue name for Armada.
-    :param job_request_items: A PodSpec that is used by Armada for submitting a job
-    :param lookout_url_template: A URL template to be used to provide users
-        a valid link to the related lookout job in this operator's log.
-        The format should be:
-        "https://lookout.armada.domain/jobs?job_id=<job_id>" where <job_id> will
-        be replaced with the actual job ID.
-    :param poll_interval: How often to poll jobservice to get status.
-    :return: an armada operator instance
+
+    template_fields: Sequence[str] = ("job_request", "job_set_prefix")
+
     """
+Initializes a new ArmadaOperator.
 
-    template_fields: Sequence[str] = ("job_request_items",)
+:param name: The name of the job to be submitted.
+:type name: str
+:param channel_args: The gRPC channel arguments for connecting to the Armada server.
+:type channel_args: GrpcChannelArgs
+:param armada_queue: The name of the Armada queue to which the job will be submitted.
+:type armada_queue: str
+:param job_request: The job to be submitted to Armada.
+:type job_request: JobSubmitRequestItem
+:param job_set_prefix: A string to prepend to the jobSet name
+:type job_set_prefix: Optional[str]
+:param lookout_url_template: Template for creating lookout links. If not specified
+then no tracking information will be logged.
+:type lookout_url_template: Optional[str]
+:param poll_interval: The interval in seconds between polling for job status updates.
+:type poll_interval: int
+:param container_logs: Name of container whose logs will be published to stdout.
+:type container_logs: Optional[str]
+:param deferrable: Whether the operator should run in a deferrable mode, allowing
+for asynchronous execution.
+:type deferrable: bool
+:param job_acknowledgement_timeout: The timeout in seconds to wait for a job to be
+acknowledged by Armada.
+:type job_acknowledgement_timeout: int
+:param kwargs: Additional keyword arguments to pass to the BaseOperator.
+"""
 
     def __init__(
         self,
         name: str,
-        armada_channel_args: GrpcChannelArgsDict,
-        job_service_channel_args: GrpcChannelArgsDict,
+        channel_args: GrpcChannelArgs,
         armada_queue: str,
-        job_request_items: List[JobSubmitRequestItem],
+        job_request: JobSubmitRequestItem,
+        job_set_prefix: Optional[str] = "",
         lookout_url_template: Optional[str] = None,
         poll_interval: int = 30,
+        container_logs: Optional[str] = None,
+        token_retriever: Optional[TokenRetriever] = None,
```

Comment:
we don't have doc for the token retriever (should it be called k8s_token_retriever?).